samples/src/App.js

```diff
@@ -32,41 +32,142 @@ function App() {
+    } else {
+      effect.disable()
+    }
+
+
+
+  }
+
   const createMicrophoneTrackAction = async (deviceId) => {
     await createMicrophoneTrack({ deviceId }).then((localMicrophoneTrack) => {
-      setLocalMicrophoneTrack(localMicrophoneTrack.getMediaStreamTrack());
+      setLocalMicrophoneTrack(localMicrophoneTrack);
+      updateEvents("Microphone track added ")
+      localMicrophoneTrack.on('muted',(event) =>{
+        updateEvents("Microphone "+ event.trackState.muted)
+      })
+
+      localMicrophoneTrack.on('ended',() =>{
+        updateEvents("Microphone ended")
+      })
+
+      localMicrophoneTrack.on("underlying-track-change",() =>{
+        updateEvents("Microphone track change")
+      })
     });
   };
 
   const createCameraTrackAction = async (deviceId) => {
-    const effect = new VirtualBackgroundEffect({
-      mode: `BLUR`,
-      blurStrength: `STRONG`,
-      quality: `HIGH`,
-    });
+
     await createCameraTrack({ deviceId }).then((localCameraTrack) => {
-      setLocalCameraTrack(localCameraTrack.getMediaStreamTrack());
+      setLocalCameraTrack(localCameraTrack);
       window.localCameraTrack = localCameraTrack;
 
+      localCameraTrack.on('muted',(event) =>{
+        updateEvents("Camera track", event.trackState.muted)
+      })
+      localCameraTrack.on('ended',(event) =>{
+        updateEvents("Camera track Ended")
+      })
+
+
       localCameraTrack.on("underlying-track-change", () => {
+        updateEvents("Camera track change ")
         console.log(localCameraTrack.getMediaStreamTrackWithEffects());
-        setLocalCameraTrack(
-          localCameraTrack.getMediaStreamTrackWithEffects() ||
-            localCameraTrack.getMediaStreamTrack()
-        );
       });
-      localCameraTrack.addEffect("blur", effect);
-      effect.enable();
     });
   };
 
   const createDisplayTrackAction = async ({ withAudio, constraints }) => {
     await createDisplayTrack({ constraints, withAudio }).then(
       ({ localDisplayTrack, localComputerAudioTrack }) => {
         const displayTracks = [];
-        displayTracks.push(localDisplayTrack.getMediaStreamTrack());
-        if (localComputerAudioTrack)
-          displayTracks.push(localComputerAudioTrack?.getMediaStreamTrack());
+        displayTracks.push(localDisplayTrack);
+        if (localComputerAudioTrack) {
+          localComputerAudioTrack.on('muted',(event) =>{
+            updateEvents("Computer Audio ", event.trackState.muted)
+          })
+          localComputerAudioTrack.on('ended',(event) =>{
+            updateEvents("Computer Audio Track Ended ")
+          })
+
+          localDisplayTrack.on("underlying-track-change",(event) =>{
+            updateEvents("Computer Audio underlying-track-change")
+          })
+          displayTracks.push(localComputerAudioTrack);
+        }
+          
+
+
+        localDisplayTrack.on('muted',() =>{
+
+          })
+          localDisplayTrack.on('ended',() =>{
+            
+          })
```

Comment:
empty? also looks like indentation is wrong

Reply:
yep need to fill it up, thought will do it later , will do it